datafusion/core/tests/sql/expr.rs

```diff
@@ -622,6 +622,17 @@ async fn test_not_expressions() -> Result<()> {
     ];
     assert_batches_eq!(expected, &actual);
 
+    let sql = "SELECT not(1), not(0)";
+    let actual = execute_to_batches(&ctx, sql).await;
```

Comment:
maybe lets move it to `scalar.slt` sqllogic test?

Reply:
I agree it would be nice to move to https://github.com/apache/arrow-datafusion/blob/main/datafusion/core/tests/sqllogictests/test_files/scalar.slt but we can do that potentially as a follow on PR

Reply:
I will port some tests over

Reply:
Follow on PR to port the tests: https://github.com/apache/arrow-datafusion/pull/5968